calculator_test.go

```diff
@@ -1,24 +1,173 @@
+		{11.2, []float64{89.8}, -78.6, "Subtracting a positive number from a smaller positive number yields a negative number"},
+		{-100, []float64{-111}, 11, "Subtracting a negative number from a larger negative number yields a positive number"},
+		{0, []float64{6, 6, 15}, -27, "Subtracting multiple positive number from zero yields a negative number"},
+		{0, []float64{-22}, 22, "Subtracting a negative number from zero yields a positive number"},
+	}
+	for _, c := range testCases {
+		got := Subtract(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a    float64
+		b    []float64
+		want float64
+		name string
+	}{
+		{1, []float64{1}, 1, "Multiplying one by one always yields one"},
+		{3999191, []float64{1, 1, 1}, 3999191, "Multiplying a number by one any number of times yields that number"},
+		{3, []float64{3}, 9, "Multiplying a number by itself equals that number squared"},
+		{-10, []float64{-10}, 100, "Multiplying two negative number yields a positive number"},
+		{-5, []float64{-12, -67}, -4020, "Multiplying three negative number yields a negative number"},
+		{12, []float64{-1.5}, -18, "Multiplying a negative number by a positive number yields a negative number"},
+		{43, []float64{0}, 0, "Multiplying any number by zero yields zero"},
+	}
+	for _, c := range testCases {
+		got := Multiply(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a             float64
+		b             []float64
+		want          float64
+		errorExpected bool
+		name          string
+	}{
+		{10, []float64{1}, 10, false, "Dividing a number by 1 yields that number"},
+		{99, []float64{1, 1, 1}, 99, false, "Dividing a number by one multiple times yields that same number"},
+		{312, []float64{0}, 0, true, "Dividing a number by zero returns an error"},
+		{50, []float64{1, 5, 0, 200}, 0, true, "Dividing a number by multiple numbers with zero as one of them returns an error"},
+		{0, []float64{0}, 0, true, "Dividing a zero by zero returns an error"},
+		{1, []float64{2, 2, 2}, 0.125, false, "Dividing a number by multiple larger numbers yields a fraction"},
+		{100, []float64{0.1, 10}, 100, false, "Dividing a number by a large number and a fraction"},
+		{100, []float64{10000000}, 0.00001, false, "Dividing a number by a much larger number yields a small fraction"},
+		{100, []float64{10000000}, 0.00001, false, "Dividing a number by a much larger number yields a small fraction"},
+	}
+	for _, c := range testCases {
+		got, err := Divide(c.a, c.b...)
+		if c.errorExpected && err == nil {
+			t.Errorf("%s: expected error, got nil", c.name)
+		}
+		if !c.errorExpected && err != nil {
+			t.Errorf("%s: expected no errors, got \"%v\"", c.name, err)
+		}
+		if c.want != got {
```

Comment:
And we only want to do this check in the non-error case, so we could short-circuit this `if` with `!errReceived`, for example.